bitseq/sequence.go

```diff
@@ -308,17 +308,17 @@ func (h *Handle) set(ordinal, start, end uint64, any bool, release bool) (uint64
 
 	for {
 		var store datastore.DataStore
-		h.Lock()
+		h.RLock()
 		store = h.store
-		h.Unlock()
+		h.RUnlock()
 		if store != nil {
-			if err := store.GetObject(datastore.Key(h.Key()...), h); err != nil && err != datastore.ErrKeyNotFound {
+			if err = store.GetObject(datastore.Key(h.Key()...), h); err != nil && err != datastore.ErrKeyNotFound {
 				return ret, err
 			}
 		}
 
-		h.Lock()
 		// Get position if available
```

Comment:
I see the following problem here.
Go routine A takes from the store objA
Go routine B takes from the store objA (using the same key)

Go routine A now modifies objA, same the Go routine B with its own copy

Now Go routine A saves its object into the store
Go routine B saves its object into the store and rewrites the obj of Go routine A

At this point the state of Go routine A is lost, I think we need a way to guarantee that 2 go routines won't fetch the same object from the store